benchmarks/Swaggatherer/SwaggathererApplication.cs

```diff
@@ -0,0 +1,250 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Microsoft.AspNetCore.Routing.Template;
+using Microsoft.Extensions.CommandLineUtils;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Swaggatherer
+{
+    internal class SwaggathererApplication : CommandLineApplication
+    {
+        public SwaggathererApplication()
+        {
+            Invoke = InvokeCore;
+
+            Input = Option("-i", "input swagger 2.0 JSON file", CommandOptionType.MultipleValue);
+            InputDirectory = Option("-d", "input directory", CommandOptionType.SingleValue);
+            Output = Option("-o", "output", CommandOptionType.SingleValue);
+
+            HelpOption("-h|--help");
+        }
+
+        public CommandOption Input { get; }
+
+        public CommandOption InputDirectory { get; }
+
+        public CommandOption Output { get; }
+
+        private int InvokeCore()
+        {
+            if (!Input.HasValue() && !InputDirectory.HasValue())
+            {
+                ShowHelp();
+                return 1;
+            }
+
+            if (Input.HasValue() && InputDirectory.HasValue())
+            {
+                ShowHelp();
+                return 1;
+            }
+
+            if (!Output.HasValue())
+            {
+                Output.Values.Add("Out.generated.cs");
+            }
+
+            if (InputDirectory.HasValue())
+            {
+                Input.Values.AddRange(Directory.EnumerateFiles(InputDirectory.Value(), "*.json", SearchOption.AllDirectories));
+            }
+
+            var entries = new List<RouteEntry>();
```

Comment:
I tried to do a good job adding comments. let me know if anything is super unclear.